src/Hal/Report/Csv/Reporter.php

```diff
@@ -40,30 +45,27 @@ public function generate(Metrics $metrics)
         if (!$logFile) {
             return;
         }
-        if (!file_exists(dirname($logFile)) || !is_writable(dirname($logFile))) {
-            throw new \RuntimeException('You don\'t have permissions to write CSV report in ' . $logFile);
+
+        $logDir = dirname($logFile);
+        if (!file_exists($logDir) || !is_writable($logDir)) {
+            throw new RuntimeException('You do not have permissions to write CSV report in ' . $logFile);
         }
 
-        $availables = (new Registry())->allForStructures();
-        $hwnd = fopen($logFile, 'w');
-        fputcsv($hwnd, $availables);
+        $availableMetrics = (new Registry())->allForStructures();
+        $logPointer = fopen($logFile, 'wb');
+        fputcsv($logPointer, $availableMetrics);
 
         foreach ($metrics->all() as $metric) {
             if (!$metric instanceof ClassMetric) {
                 continue;
             }
-            $row = [];
-            foreach ($availables as $key) {
+            $row = array_map(static function ($key) use ($metric) {
                 $data = $metric->get($key);
-                if (is_array($data) || !is_scalar($data)) {
-                    $data = 'N/A';
-                }
+                return (!is_scalar($data)) ? 'N/A': $data;
```

Comment:
why removing the condition `is_array` ? Why the objective of this refactoring ?

Reply:
Well, `is_array()` and `!is_scalar()` are tautologies: if it is an array, it is definitly not a scalar, and if it is not an array, the refactored condition is the same than the previous one.